src/xmpp/iq.c

```diff
@@ -2434,3 +2445,12 @@ _iq_free_affiliation_set(ProfPrivilegeSet *affiliation_set)
         free(affiliation_set);
     }
 }
+
+static void
+_iq_free_affiliation_list(ProfAffiliationList *affiliation_list)
+{
+    if (affiliation_list) {
+        free(affiliation_list->affiliation);
+        free(affiliation_list);
+    }
```

Comment:
Should we set affiliation_list to NULL afterwards? Is it possible that we will get here again?